### ISSUE TITLE
feat: AI translated voiceover / Port Voice Over Translation userscript

### ISSUE THREAD

**Issue by Metadorius** [ReVanced/revanced-patches#2667]
### Type

Functionality

### Issue

ReVanced app lacks AI voiceover feature which is available using userscripts on desktop.

### Feature

Implement/port Voice Over Translation userscript which uses Yandex services under the hood in order to translate the voice from one language to another. You can see more details here: https://github.com/ilyhalight/voice-over-translation/blob/master/README-EN.md (contrary to the readme the plugin can actually translate either way, not just to Russian).

We've asked the author @ilyhalight in https://github.com/ilyhalight/voice-over-translation/issues/187 but he doesn't know Kotlin, although he also admits it would be great to have this feature integrated in ReVanced.

### Motivation

Despite English being top-1 international language, there are still many people who don't know English, and also there's a lot of non-English videos. It's considerably more tedious to watch videos in foreign language for people who don't know the language. For instance, I do, but many of my local friends and family members don't know and they would still benefit a lot from that, like my mom who only has her phone and can't use the userscript to translate the video). It would help greately to have this integrated in ReVanced to make videos more accessible worldwide.

**Comment by oSumAtrIX** [comment]
Is this about YouTube? Then please add it to the issue.

**Comment by Metadorius** [comment]
> Is this about YouTube? Then please add it to the issue.

Done, didn't know ReVanced is not only YT now (?).

**Comment by bropines** [comment]
@oSumAtrIX Sorry for the tag, but will this functionality be added, or is it not worth waiting for?

**Comment by masterwishx** [comment]
Also interested in this future

**Comment by Spp595** [comment]
For me, this is the only reason to occasionally use the mobile-browser version of YouTube.
